add documentation for validator (#233)
openvinotoolkit/datumaro: docs/user_manual.md

</details>

### Validate project annotations

> This command inspects annotations based on the task type

Reviewer: ```suggestion
This command inspects annotations with respect to the task type
```

and stores the result in JSON file.

The task types supported are `classification`, `detection`, and `segmentation`.

The validation result contains
- annotation statistics based on the task type
- validation reports, such as
    - items not having annotations
    - items having undefined annotations
    - imbalanced distribution in class/attributes

>     - too small or large values (items_far_from_mean)

Reviewer: `items_far_from_mean` - it's not clear, what it means here.

Author: `items_far_from_mean` is a keyword of `numerical_stat_template`.
It means that a value in distribution is too big or too small to the avaerage value(=mean) of the distribution.
I think it's better to remove `(items_far_from_mean)`.

- summary

Usage:

``` bash
datum validate --help

datum validate -p <project dir> <task_type>
```

Validation Result:

<details>

``` bash
{
    'statistics': {
        ## common statistics
        'label_distribution': {
            'defined_labels': <dict>,   # <label:str>: <count:int>
            'undefined_labels': <dict>
            #  <label:str>: {
            #    'count': <int>,

>             #    'items_with_undefined_label': [(<item.id>, <item.subset>), ]

Reviewer: Please be consistent in field definitions. 
It is unclear, what the dot means and what type the fields have.

Author: `item` means  `DatasetItem`. I'll correct this.

            #  }
        },
        'attribute_distribution': {
            'defined_attributes': <dict>,
            #  <label:str>: {
            #    <attribute:str>: {
            #      'distribution': {<attr_value:str>: <count:int>, },
            #      'items_missing_attribute': [(<item.id>, <item.subset>), ]
            #    }
            #  }
            'undefined_attributes': <dict>
            #  <label:str>: {
            #    <attribute:str>: {
            #      'distribution': {<attr_value:str>: <count:int>, },
            #      'items_with_undefined_attr': [(<item.id>, <item.subset>), ]
            #    }
            #  }
        },
        'total_ann_count': <int>,
        'items_missing_annotation': <list>, # [(<item.id>, <item.subset>), ]

        ## statistics for classification task
        'items_with_multiple_labels': <list>, # [(<item.id>, <item.subset>), ]

        ## statistics for detection task
        'items_with_invalid_value': <dict>,
        #  '(<item.id>, <item.subset>)' : {
        #    <ann.id> : [ <property:str>, ]
        #  }
        #  properties: 'x', 'y', 'width', 'height',

>         #              'area(wxh)', 'ratio(w/h)', 'short', 'long'

Reviewer: Clarify the meaning of `short` and `long`

Author: OK. I will.

        #    <ann.id> : { <'width'|'height'>: <value>, }
        #  }
        'bbox_distribution_in_label': <dict>, # <label:str>: <bbox_template>
        'bbox_distribution_in_attribute': <dict>,
        #  <label:str>: {
        #     <attribute:str>: { <attr_value>: <bbox_template>, }
        #  }
        'bbox_distribution_in_dataset_item': <dict>,
        # '(<item.id>, <item.subset>)': <bbox count: int>

        ## statistics for segmentation task
        'items_with_invalid_value'] = <dict>,
        #  '(<item.id>, <item.subset>)' : {
        #    <ann.id> : [ <property:str>, ]
        #  }
        #  properties: 'area', 'width', 'height'
        'mask_distribution_in_label'] = <dict>, # <label:str>: <mask_template>
        'mask_distribution_in_attribute'] = <dict>,
        #  <label:str>: {
        #     <attribute:str>: { <attr_value>: <mask_template>, }
        #  }
        'mask_distribution_in_dataset_item'] = <dict>,
        # '(<item.id>, <item.subset>)': <mask/polygon count: int>
    },
    'validation_reports': <list>, #[ <validation_error_format>, ]
    # validation_error_format = {
    #   'anomaly_type': <str>,  # see datumaro/components/errors.py

>     #   'description': <str>,   # see datumaro/components/errors.py

Reviewer: Who is supposed to read the code?

Author: Do you mean the anomaly_type should be documented, instead of referring to the code?

Author: I thought that this information is needed to parse the validation result.

    #   'severity': <str>, # 'warning' or 'error'    #

>     #   'item_id': <str>,  # when the validation target is a DatasetItem

Reviewer: Validation target is a project or a dataset. Find a better phrasing.

Author: The validator inspects each DatasetItem and the distribution of them.
Here, is it assumed that the user doesn't know what the 'DatasetItem' is?
